src/toxcore/dht_new/client.rs

```diff
@@ -0,0 +1,120 @@
+/*
+    Copyright © 2017 Zetok Zalbavar <zexavexxe@gmail.com>
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+/*!
+Hold peer info. connected.
+The object of this struct is an object per a peer.
+*/
+
+use futures::{Sink, Future};
+use futures::sync::mpsc;
+use tokio_io::IoFuture;
+
+//use std::collections::VecDeque;
+use std::io::{ErrorKind, Error};
+use std::net::SocketAddr;
+
+use toxcore::crypto_core::*;
+use toxcore::dht_new::packet::*;
+use toxcore::dht_new::packed_node::*;
+use toxcore::dht_new::codec::*;
+
+/// Shorthand for the transmit half of the message channel.
+type Tx = mpsc::UnboundedSender<DhtUdpPacket>;
+
+/// peer info.
+#[derive(Clone, Debug)]
+pub struct Client {
+    /// Public key of peer
+    pub pk: PublicKey,
+    /// socket address of peer
+    pub addr: SocketAddr,
+    /// tx split of channel to send packet to this peer via udp socket
+    pub tx: Tx,
```

Comment:
I would say the server struct should store tx but not the client since it's shared between all clients. And the same for send/send_to methods.